### DISCUSSION TITLE
Sidebar arrow for empty folders

### DISCUSSION THREAD

**Discussion by Ishani-Chatterjee** [dbeaver/dbeaver#13470] [General]
**Problem I face in sidebar**
 I feel that when a folder is empty an arrow should not be shown.
When I click on it only then the arrow disappears.

**What is happening:**
Even If the folder is empty there is an arrow and when I click on it I find the folder is empty, and only then does the arrow disappears

**What is expected (I feel):**
If a folder is empty there should not be any arrow.

**Why is this a problem:**
Because there are so many folders and If we need to check which of them is empty, we need to click all the arrows.

Like in my case the Procedures folder is empty, but the arrow is shown. When I click on the arrow the arrow disappears.
![image](https://user-images.githubusercontent.com/59927726/128534252-58306c9b-2432-4f34-9c3c-ce2c4d7fbadd.png)
![image](https://user-images.githubusercontent.com/59927726/128534358-2f792d25-82c4-479c-b019-a809823b67ec.png)

**Comment by ShadelessFox** [comment]
Folder's children are loaded on demand when you expand it, so you can't know whether a given folder contains something or not.
